app/src/components/generative/ToolChoiceSelector.tsx

```diff
@@ -52,7 +52,7 @@ export function ToolChoicePicker({
   toolNames,
 }: ToolChoicePickerProps) {
   const currentKey =
-    typeof choice === "string"
+    choice === undefined || typeof choice === "string"
```

Comment:
even though we've typed this to be undefined i would do choice == null since it will catch null or undefined in case the type changes, this is still caught